tests/regression/README.md

```diff
@@ -0,0 +1,77 @@
+# Regression Testing for s2n-tls
+
+This folder contains regression tests and benchmarking tools for the `s2n-tls` library. The tests focus on various aspects of TLS connections, including handshakes and session resumptions.
+
+## Contents
+
+1. **Regression Harnesses**
+   - **config_create.rs**: Creates a minimal s2n-tls configuration.
+   - **config_configure.rs**: Configures an s2n-tls config with security policies and certificate key pairs.
+
+2. **Cargo.toml**
+   - The configuration file for building and running the regression tests using Cargo.
+
+3. **run_harnesses.sh**
+   - Script to run all harnesses, a specified harness, or a combination of harnesses with Valgrind and store annotated results.
+
+
+## Prerequisites
+
+Ensure you have the following installed:
+- Rust (with Cargo)
+- Valgrind (for crabgrind instrumentation)
+
+## Running the Harnesses with Valgrind
+To run the harnesses with Valgrind and store the annotated results, use the `run_harnesses.sh` script:
+
+### Make the Script Executable
+
+Make the `run_harnesses.sh` script executable:
```

Comment:
Let's just go ahead and make it executable in the `git` repository so that people don't have to manually do this.